src/core/server/plugins/plugins_system.ts

```diff
@@ -158,13 +158,48 @@ export class PluginsSystem {
         timeout: 30 * Sec,
         errorMessage: `Start lifecycle of "${pluginName}" plugin wasn't completed in 30sec. Consider disabling the plugin and re-start.`,
       });
-
       contracts.set(pluginName, contract);
     }
+    await this.healthCheckOpenSearchPlugins(deps);
 
     return contracts;
   }
 
+  private async healthCheckOpenSearchPlugins(deps: PluginsServiceStartDeps) {
+    // make _cat/plugins?format=json call to the OpenSearch instance
+    const opensearchPlugins = await this.getOpensearchPlugins(deps);
+    for (const pluginName of this.satupPlugins) {
+      this.log.debug(`For plugin "${pluginName}"...`);
+      const plugin = this.plugins.get(pluginName)!;
+      const pluginBackendDeps = new Set([...plugin.requiredOpenSearchPlugins]);
+      pluginBackendDeps.forEach((opensearchPlugin) => {
+        if (!opensearchPlugins.find((obj) => obj.component === opensearchPlugin)) {
+          this.log.error(
+            `OpenSearch plugin "${opensearchPlugin}" is not installed on the cluster for the OpenSearch Dashboard plugin to function as expected.`
+          );
+          throw new Error(
+            `Plugin "${pluginName}" has a dependency on OpenSearch plugin "${opensearchPlugin}" which is not installed on the cluster.`
+          );
```

Comment:
We should allow plugins to gracefully degrade by default and only throw if the plugin says they cannot live without their counterpart.

Reply:
This is only when plugins choose to explicitly define backend dependencies as a required dependency in the plugin manifest config for `requiredOpensearchPlugins` so I think it is fair to throw error in this scenario.

Reply:
This might have been my fault to think about throwing error.

But I do see the case by @AMoo-Miki. Thinking from an extensibility some plugin might define a required OpenSearch plugin that doesn't exist, either by mistake or on purpose, and then OpenSearch Dashboards might not start.

i'm in the middle ground where we can either not throw the error and just log it for now. Or maybe allow for dynamically configuring as a cluster admin to let OpenSearch Dashboards start up if missing OpenSearch plugins.

Something similar to `opensearch.ignoreVersionMismatch`